pkgs/build-support/kernel/compress-firmware.nix

```
@@ -1,24 +1,38 @@
-{ runCommand, lib }:
+{ runCommand, lib, type ? "zstd", zstd }:
 
 firmware:
 
 let
+  compressor = {
+    xz = {
+      ext = "xz";
+      nativeBuildInputs = [ ];
+      cmd = file: target: ''xz -9c -T1 -C crc32 --lzma2=dict=2MiB "${file}" > "${target}"'';
+    };
+    zstd = {
+      ext = "zst";
+      nativeBuildInputs = [ zstd ];
+      cmd = file: target: ''zstd -T1 -6 --check -f "${file}" -o "${target}"'';
```

```suggestion
      cmd = file: target: ''zstd -T1 -19 --long --check -f "${file}" -o "${target}"'';
```
You can go as high as level 19, and `--long` should work too. This is consistent with what is configured for `xz`.

---

I wonder if we could reuse/expand the data from `pkgs/build-support/kernel/initrd-compressor-meta.nix`?